core/src/main/java/org/wildfly/plugin/core/DeploymentOperations.java

```diff
@@ -0,0 +1,479 @@
+ * managed domain. If the server groups are {@linkplain Set#isEmpty() is empty} the standalone deployment operations
+ * will be created. Otherwise deployment operations for managed domains will be created.
+ * </p>
+ * <p>
+ * All operations create will be composite operations for consistency of parsing the result of executing the operation.
+ * </p>
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+@SuppressWarnings({"unused", "StaticMethodOnlyUsedInOneClass", "WeakerAccess"})
+public class DeploymentOperations {
+    private static final String ENABLE = "enable";
+    private static final String ENABLED = "enabled";
+    static final ModelNode EMPTY_ADDRESS = new ModelNode().setEmptyList();
+
+    static {
+        EMPTY_ADDRESS.protect();
+    }
+
+    /**
+     * Creates an {@linkplain ModelNode address} that can be used as the address for an operation. The address is
+     * simply a {@link ModelNode} of type {@link ModelType#LIST}.
+     * <p>
+     * The string is split into key/value pairs. If the final key does not have a value an {@code *} is used to
+     * indicate a  wildcard for the address.
+     * </p>
+     *
+     * @param pairs the key/value pairs to use
+     *
+     * @return an address for the key/value pairs
+     */
+    public static ModelNode createAddress(final String... pairs) {
+        return createAddress(Arrays.asList(pairs));
+    }
+
+    /**
+     * Creates an {@linkplain ModelNode address} that can be used as the address for an operation. The address is
+     * simply a {@link ModelNode} of type {@link ModelType#LIST}.
+     * <p>
+     * The string is split into key/value pairs. If the final key does not have a value an {@code *} is used to
+     * indicate a  wildcard for the address.
+     * </p>
+     *
+     * @param pairs the key/value pairs to use
+     *
+     * @return an address for the key/value pairs
+     */
+    public static ModelNode createAddress(final Iterable<String> pairs) {
+        final ModelNode address = new ModelNode();
+        final Iterator<String> iterator = pairs.iterator();
+        while (iterator.hasNext()) {
+            final String key = iterator.next();
+            final String value = (iterator.hasNext() ? iterator.next() : "*");
+            address.add(key, value);
+        }
+        return address;
+    }
+
+    /**
+     * Creates an operation to add deployment content to the deployment repository.
+     * <p>
+     * Note this does not deploy the content.
+     * </p>
+     *
+     * @param content     the path to the content to deploy
+     * @param name        the name for the deployment content, if {@code null} the paths file name is used
+     * @param runtimeName the runtime name for the deployment content
+     *
+     * @return the add operation
+     */
+    public static Operation createDeploymentAddOperation(final Path content, final String name, final String runtimeName) {
```

Comment:
It's here for convenience only. It's useful in cases where you just want to add content, but not deploy it. This may be more common in a managed domain where you just want to add content to the repository, but not add it to any server groups.